libstuff/libstuff.cpp

```diff
@@ -1854,6 +1857,17 @@ bool S_recvappend(int s, string& recvBuffer) {
     int flags = fcntl(s, F_GETFL);
     bool blocking = !(flags & O_NONBLOCK);
 
+    // Log size of the buffer before we read from it.
+    int bytesInBuffer = 0;
+    int ret = 0;
+    ret = ioctl(s, FIONREAD, &bytesInBuffer);
+
+    if (ret < 0) {
+        SHMMM("Unable to get length of socket buffer error: " << strerror(S_errno));
+    } else {
+        SINFO("[performance] " << bytesInBuffer << " bytes in the socket buffer before receiving.");
```

Comment:
Is this going to like insanely blow up the logs with extremely high frequency logging?

Reply:
This should be fine, we've added similar log lines before with no ill effects and they're filtered out of elastic search anyway.

Reply:
We have a similar line in send(), it is somewhat noisy, but as tyler said these are filtered out of our ES cluster, so it will only slightly increase the on disk logging size.